service/controller/appcatalog/v1/resource/index/resource.go

```diff
@@ -39,14 +37,6 @@ type Resource struct {
 	projectName    string
 }
 
-func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
-	return nil, nil
-}
-
-func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
-	return nil
-}
-
 // New creates a new configured index resource.
 func New(config Config) (*Resource, error) {
```

Comment:
I was thinking to improve the docs here a bit. Do you think it makes sense to have descriptions and pictures about the purpose and functionality of the resource implementations? Imagine you just join the company and check the operators. It would be cool to provide everyone with more information on **how** our software works.

Reply:
Definitely yes, pictures are much needed to understand how operatorkit is working. maybe I could create related issue?

Reply:
Yes the docs situation needs to improve. I was thinking we could have architecture docs for each operator in https://github.com/giantswarm/architecture.

We can then use the resource descriptions from the doc in the operator resources as godoc. WDYT? 

@tomahawk28 Could you start creating an app-operator doc?

I'd like to use the same structure as for cluster-operator. See https://github.com/giantswarm/giantswarm/pull/5405.